packages/flutter_tools/lib/src/host_validator.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+
+import 'doctor.dart';
+import 'globals.dart';
+
+/// Verify that the required hosts ( they are stored at [_requiredHosts] ) are reachable
+class HostValidator extends DoctorValidator {
+  HostValidator() : super('Host availability');
+
+  @override
+  String get slowWarning => 'Host availability check is taking a long time...';
+
+  http.Client httpClient = http.Client();
```

Comment:
The http.Client should be injected in the constructor so that we can avoid making network requests during the test